crypto/tls/tls.go

```diff
@@ -13,11 +13,11 @@ import (
 
 // ClientConfig is the config for client TLS.
 type ClientConfig struct {
-	CertPath           string `yaml:"tls_cert_path"`
-	KeyPath            string `yaml:"tls_key_path"`
-	CAPath             string `yaml:"tls_ca_path"`
-	ServerName         string `yaml:"tls_server_name"`
-	InsecureSkipVerify bool   `yaml:"tls_insecure_skip_verify"`
+	CertPath           string `yaml:"tls_cert_path" category:"advanced"`
```

Comment:
I see some discrepancy. The `EnableTLS` is basic (eg. in grpcclient or etcd) but the settings for TLS are not. Either TLS is basic (and then all paths to cert/key/CA are basic as well) or the whole TLS setup is advanced, but I think having a mix is inconsistent.

Reply:
@pracucci I categorized all `tls_enabled` parameters as advanced in the meantime. Is there anything else that needs categorizing as well?

Reply:
TLS config should have been fully addressed in this PR 👍